### ISSUE TITLE
Did not get confirmation UiAutomator2 deleteSession worked; Error was: UnknownError: An unknown server-side error occurred while processing the command. Original error: Trying to proxy a session command without session id

### ISSUE THREAD

**Issue by risinek** [appium/appium#11127]
## The problem

Appium tests started failing on Mac Mini few weeks ago (maybe months already) with last log error message: 

```
Unable to remove port forward 'Error executing adbExec. Original error: 'Command '/usr/local/Caskroom/android-sdk/3859397\,26.0.2/platform-tools/adb -P 5037 -s 5200fc8bf01b9441 forward --remove tcp\:8200' exited with code 1'; Stderr: 'error: listener 'tcp:8200' not found'; Code: '1''
```

```
Did not get confirmation UiAutomator2 deleteSession worked; Error was: UnknownError: An unknown server-side error occurred while processing the command. Original error: Trying to proxy a session command without session id
```

I tried to reinstall appium server, tried update to 1.8.1 and then also beta, but with no luck, still same issue. 
When the test fails with this error log, I can see _com.appium.uiautomator2.server_ is still running in process list. Not sure if this may be the cause. Also I noticed, that if I run 

> adb -P 5037 -s 5200fc8bf01b9441 shell kill -9

with PID of that remaining uiautoamtor2 server manually after the test fails, I get permission denied error. I saw this command in logs, with the same permission denied error message. But I have no clue, why is it missing permissions and what I can do to fix it..  

## Environment

* Appium version (or git revision) that exhibits the issue: 1.7.2, 1.8.1, 1.8.2 beta
* Last Appium version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Appium:
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G65
* Node.js version (unless using Appium.app|exe):8.0.0
* Mobile platform/version under test: Android
* Real device or emulator/simulator: Real
* Appium CLI or Appium.app|exe: CLI

## Details
I'm running the appium server with 

> appium --relaxed-security --log-timestamp

## Link to Appium logs

https://gist.github.com/risinek/5f19ee4c7600a83cad1409da9e8e8bab

**Comment by mykola-mokhnach** [comment]
From the log it looks like two different drivers are trying to setup their sessions on the same device in parallel. I would assume there is a problem in the client code

**Comment by risinek** [comment]
I use serenity to run the tests (if it may be somehow related).. If I run the appium server on my Mac Air and run the exact same client (same repo), it works fine.. But when I run it on remote Mac Mini, I get these errors. Only difference in setup is, that the Mac Mini has plenty of Android devices connected to it directly via USB, so I have to use udid capability to choose one. If I run the client against appium server on localhost, I have just one device connected, hence I don't use udid capability. (I saw this may affect the results on another issue https://github.com/appium/appium/issues/8234#issuecomment-380030691)

**Comment by risinek** [comment]
Well, I tried on another device.. Got this:
```2018-07-30 15:11:32:940 - [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8200/wd/hub/status] with no body
2018-07-30 15:11:33:022 - [debug] [UiAutomator2] [UIAutomator2] java.lang.SecurityException: Permission Denial: starting instrumentation ComponentInfo{io.appium.uiautomator2.server.test/android.support.test.runner.AndroidJUnitRunner} from pid=17807, uid=17807 not allowed because package io.appium.uiautomator2.server.test does not have a signature matching the target io.appium.uiautomator2.server
2018-07-30 15:11:33:022 - [debug] [UiAutomator2] [UIAutomator2] 	at android.os.Parcel.readException(Parcel.java:1958)
2018-07-30 15:11:33:022 - [debug] [UiAutomator2] [UIAutomator2] 	at android.os.Parcel.readException(Parcel.java:1904)
2018-07-30 15:11:33:022 - [debug] [UiAutomator2] [UIAutomator2] 	at android.app.IActivityManager$Stub$Proxy.startInstrumentation(IActivityManager.java:5586)
2018-07-30 15:11:33:022 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.commands.am.Instrument.run(Instrument.java:408)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.commands.am.Am.runInstrument(Am.java:232)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.commands.am.Am.onRun(Am.java:125)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.internal.os.BaseCommand.run(BaseCommand.java:54)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.commands.am.Am.main(Am.java:95)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:286)
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] INSTRUMENTATION_STATUS: id=ActivityManagerService
2018-07-30 15:11:33:023 - [debug] [UiAutomator2] [UIAutomator2] INSTRUMENTATION_STATUS: Error=Permission Denial: starting instrumentation ComponentInfo{io.appium.uiautomator2.server.test/android.support.test.runner.AndroidJUnitRunner} from pid=17807, uid=17807 not allowed because package io.appium.uiautomator2.server.test does not have a signature matching the target io.appium.uiautomator2.server
2018-07-30 15:11:33:024 - [debug] [UiAutomator2] [UIAutomator2] INSTRUMENTATION_STATUS_CODE: -1```

**Comment by risinek** [comment]
Well `--session-override` Appium server parameter seems to helped me here.. Tests are passing now, but I will do some more runs to be sure, it is fixed

**Comment by neogeek13** [comment]
FWIW, in case it helps anyone else. rolling back from 1.8.1 to 1.8.0 fixed it for me. Installed 1.8.1 and testing again and it was still working so... maybe something got fouled up somehow in my install.

**Comment by SuryakiranP** [comment]
![Capture111111](https://user-images.githubusercontent.com/51612593/60506443-77077f00-9ce3-11e9-8d4e-e5107691fa75.PNG)

**Comment by SuryakiranP** [comment]
Please, can anyone provide me the solution for the above?

**Comment by SuryakiranP** [comment]
I resolved my issue by reinstalling appium.

**Comment by suaadBatis** [comment]
The same issue I thinks I have to reinstall the appium

**Comment by gi865** [comment]
i reinstalled appium,but not working

**Comment by suaadBatis** [comment]
@gi865  you have to look for if you have tools  folder inside SDK[android_sdk/tools/bin/] if not  exist you have to download it manually then manage your path

**Comment by Nisim761** [comment]
> Well `--session-override` Appium server parameter seems to helped me here.. Tests are passing now, but I will do some more runs to be sure, it is fixed

Did it solve the issue?

**Comment by kolgenpav** [comment]
For me the issue was resolved after tested phone restart.

**Comment by zzttff** [comment]
I resolved my issue by reboot phone.

**Comment by joshuapinter** [comment]
> For me the issue was resolved after tested phone restart.

Damn, that actually worked. Thanks!
